tests/converters/test_base_converter.py

```diff
@@ -0,0 +1,244 @@
+    converter = BaseConverter()
+    assert converter.convert({}) == {}
+
+    converter._schema = ContextModuleSchema()  # pylint: disable=protected-access
+    context_module = {}
+
+    # When the event dict (context_module) is not valid (raises ValidationError) we return None
+    assert converter.convert(context_module) is None
+    context_module["display_name"] = "display_name_value"
+    assert converter.convert(context_module) is None
+
+    # When the _schema contains more fields than we have defined, Attribute error is raised
+    context_module[
+        "usage_key"
+    ] = "usage_key_value"  # Now context_module is a valid event
+    with pytest.raises(
+        AttributeError,
+        match="'BaseConverter' object has no attribute '(display_name|usage_key)'",
+    ):
+        converter.convert(context_module)
+
+    converter.display_name = GoTo(["foo"])
+    # Now only usage_key is missing
+    with pytest.raises(
+        AttributeError, match="'BaseConverter' object has no attribute 'usage_key'"
+    ):
+        converter.convert(context_module)
+    converter.usage_key = GoTo(["bar"])
+
+    # When a field is found the default is it's value
+    assert converter.convert(context_module) == {
+        "foo": "display_name_value",
+        "bar": "usage_key_value",
+    }
+
+    # We should be able to skip a field
+    converter.usage_key = GoTo(None)
+    assert converter.convert(context_module) == {"foo": "display_name_value"}
+
+    # We may indicate a static value for the field (field present)
+    converter.display_name = GoTo(["foo"], "static_value")
+    assert converter.convert(context_module) == {"foo": "static_value"}
+
+    # We may apply a transformation to the field
+    converter.display_name = GoTo(["foo"], lambda x: x.upper())
+    assert converter.convert(context_module) == {"foo": "DISPLAY_NAME_VALUE"}
+
+    # We may apply a transformation to the field with the help of other fields
+    converter.display_name = GoTo(
+        ["foo"],
+        Link(
+            [KwargPath("usage_key_value", ["usage_key"])],
+            lambda x, usage_key_value: (x, usage_key_value),
+        ),
+    )
+    assert converter.convert(context_module) == {
+        "foo": ("display_name_value", "usage_key_value")
+    }
+
+    # We may alter the destination path depending on the field value
+    converter.display_name = GoTo(lambda x: ["baz"] if x == "change" else ["foo"])
+    assert converter.convert(context_module) == {"foo": "display_name_value"}
+    context_module["display_name"] = "change"
+    assert converter.convert(context_module) == {"baz": "change"}
+
+    # We may alter the destination path depending on other fields values
+    context_module["display_name"] = "display_name_value"
+    converter.display_name = GoTo(
+        Link(
+            [KwargPath("usage_key_value", ["usage_key"])],
+            lambda x, usage_key_value: [x, usage_key_value],
+        )
+    )
```

Comment:
Some suggestions to change the name of the Link class

```suggestion
    converter.display_name = GoTo(
        With(
            [KwargPath("usage_key_value", ["usage_key"])],
            lambda x, usage_key_value: [x, usage_key_value],
        )
    )
```
```suggestion
    converter.display_name = GoTo(
        Dependency(
            [KwargPath("usage_key_value", ["usage_key"])],
            lambda x, usage_key_value: [x, usage_key_value],
        )
    )
```
```suggestion
    converter.display_name = GoTo(
        Including(
            [KwargPath("usage_key_value", ["usage_key"])],
            lambda x, usage_key_value: [x, usage_key_value],
        )
    )
```
```suggestion
    converter.display_name = GoTo(
        Attach(
            [KwargPath("usage_key_value", ["usage_key"])],
            lambda x, usage_key_value: [x, usage_key_value],
        )
    )
```